gdnative-core/src/core_types/geom/basis.rs

```diff
@@ -247,9 +247,37 @@ impl Basis {
         copy
     }
 
+    /// Returns linear interpolation on a sphere between two basis by weight amount (on the range of 0.0 to 1.0).
+    #[inline]
+    pub fn slerp(&self, other:Basis, weight:f32) -> Self {
+        #[inline]
+        fn lerp_f(from:f32, to:f32, weight:f32) -> f32 {
+            from + (to - from) * weight
+        }
+        let from = self.to_quat();
+        let to = other.to_quat();
+        let mut result = Basis::from_quat(from.slerp(to, weight));
+        result.elements[0] *= lerp_f(self.elements[0].length(), other.elements[0].length(), weight);
+        result.elements[1] *= lerp_f(self.elements[1].length(), other.elements[1].length(), weight);
+        result.elements[2] *= lerp_f(self.elements[2].length(), other.elements[2].length(), weight);
+
+        result
+    }
+
+    /// Returns linear interpolation between two basis by weight amount (on the range of 0.0 to 1.0).
+    #[inline]
+    pub fn linear_interpolate(&self, other:Basis, weight:f32) -> Self {
+        // this is how godot is doing it at https://github.com/godotengine/godot/blob/master/core/math/basis.cpp#L964
+        // but Godot engine output for me differs than godot-rust
+        let a = self.elements[0].linear_interpolate(other.elements[0], weight);
+        let b = self.elements[1].linear_interpolate(other.elements[1], weight);
+        let c = self.elements[2].linear_interpolate(other.elements[2], weight);
+        Basis::from_rows(a, b, c)
+    }
+
     /// Transposes the matrix.
     #[inline]
-    fn transpose(&mut self) {
```

Comment:
These methods were deliberately private, in favor of non-mutating ones like `transposed()`, for these reasons:
* The API maps GDScript directly, which typically doesn't have the mutating methods, either.
* It guides the user toward a more "functional API" without side effects, which is often easier to reason about and to test.
* Ambiguities with English verbs/nouns are avoided, e.g. "scale" (does it scale or return the scale?), "center", etc.
* With Rust's move semantics and inlining, this is likely not even slower than in-place mutations.

So I think it makes sense if those methods stay private for now, and new ones follow the same pattern. You can keep the mutating methods, maybe just remove the `pub` 🙂